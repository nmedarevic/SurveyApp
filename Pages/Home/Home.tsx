import * as React from 'react';
import {Button, StyleSheet, Text, View} from 'react-native';

export const Home = ({navigation}: any) => {
  return (
    <View
      style={[
        styles.container,
        {
          flexDirection: 'column',
        },
      ]}>
      <Text style={styles.label}>{'Welcome to Survey app'}</Text>
      <Button
        title="Start a survey"
        onPress={() => navigation.navigate('Survey', {name: 'Jane'})}
      />
      <Button
        title="View previous surveys"
        onPress={() => navigation.navigate('SurveyList', {name: 'Jane'})}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    justifyContent: 'space-around',
    alignContent: 'center',
    alignItems: 'center',
  },
  label: {
    textAlign: 'center',
    marginBottom: 10,
    fontSize: 24,
  },
});
