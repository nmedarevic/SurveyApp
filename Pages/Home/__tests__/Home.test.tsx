import React from 'react';
import renderer from 'react-test-renderer';
import {Home} from '../Home';
import {render, fireEvent} from '@testing-library/react-native';

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should initiate navigation to other pages', async () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  const {findByText} = render(<Home {...props} />);

  const surveyButton = await findByText('Start a survey');
  const listButton = await findByText('View previous surveys');

  expect(surveyButton).toBeTruthy();
  expect(listButton).toBeTruthy();

  fireEvent.press(surveyButton);

  expect(props.navigation.navigate).toHaveBeenCalledWith('Survey');
});
