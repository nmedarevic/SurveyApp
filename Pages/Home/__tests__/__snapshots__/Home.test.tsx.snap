// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Survey app
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start a survey
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View previous surveys
      </Text>
    </View>
  </View>
</View>
`;
